src/import/pull-common.c

```diff
@@ -637,6 +641,99 @@ int pull_job_restart_with_sha256sum(PullJob *j, char **ret) {
         return 1;
 }
 
+int signature_style_from_url(const char *url, SignatureStyle *ret) {
+        _cleanup_free_ char *last = NULL;
+        int r;
+
+        assert(url);
+        assert(ret);
+
+        /* Determines which kind of signature style is appropriate for this url */
+
+        r = import_url_last_component(url, &last);
+        if (r < 0)
+                return r;
+
+        if (streq(last, "SHA256SUMS.gpg")) {
+                *ret = SIGNATURE_GPG_PER_DIRECTORY;
+                return 0;
+        }
+
+        if (streq(last, "SHA256SUMS.asc")) {
+                *ret = SIGNATURE_ASC_PER_DIRECTORY;
+                return 0;
+        }
+
+        if (endswith(last, ".sha256.gpg")) {
+                *ret = SIGNATURE_GPG_PER_FILE;
+                return 0;
+        }
+
+        if (endswith(last, ".sha256.asc")) {
+                *ret = SIGNATURE_ASC_PER_FILE;
+                return 0;
+        }
```

Comment:
this is soooo close to what so this really is really close to hat same thing as is_signature_file() is doing.

maybe rename is_signature_file() to `signature_style_from_filename()`, so that it already returns the right style here. and then make `signature_style_from_url()` just a small helper that combines `import_url_last_component()` and `signature_style_from_url()`

Reply:
and any calls that so far did is_signature_file() would then just become `signature_style_from_filename(…) >= 0`